libs/core/src/lib/time/time-column/time-column.component.ts

```diff
@@ -0,0 +1,364 @@
+    @Output()
+    activeItemChange: EventEmitter<any> = new EventEmitter<any>();
+
+    /** Event emitted, when previous column should be focused */
+    @Output()
+    focusPreviousColumn: EventEmitter<void> = new EventEmitter<void>();
+
+    /** Event emitted, when next column should be focused */
+    @Output()
+    focusNextColumn: EventEmitter<void> = new EventEmitter<void>();
+
+    /** Event emitted, when certain column is activated */
+    @Output()
+    activeStateChange: EventEmitter<void> = new EventEmitter<void>();
+
+    /** Event emitted, when focus is lost from last button of column. It helps with focus trap */
+    @Output()
+    lastButtonTabKeyDown: EventEmitter<KeyboardEvent> = new EventEmitter<KeyboardEvent>();
+
+    /** @hidden */
+    @ViewChild(CarouselDirective)
+    carousel: CarouselDirective;
+
+    /** @hidden */
+    @ViewChildren(CarouselItemDirective)
+    items: QueryList<CarouselItemDirective>;
+
+    /* Whether the action bar also has a back button. */
+    @HostBinding('class.fd-time__col')
+    fdTimeColClass: boolean = true;
+
+    /**
+     * Time to wait in milliseconds after the last keydown before focusing or selecting option based on numeric/alpha
+     * keys.
+     */
+    typeaheadDebounceInterval: number = 750;
+
+    config: CarouselConfig;
+    currentIndicatorId: string = this.id + '-current-indicator';
+
+    /** @hidden */
+    private _queryKeyDownEvent: Subject<string> = new Subject<string>();
+
+    /** @hidden */
+    private _activeItem: any;
+
+    /** @hidden */
+    private _activeCarouselItem: CarouselItemDirective;
+
+    /** @hidden */
+    private _isDragging: boolean = false;
+
+    /** @hidden */
+    private _initialised: boolean = false;
+
+    /** @hidden */
+    private _subscriptions: Subscription = new Subscription();
+
+    constructor(
+        private _changeDetRef: ChangeDetectorRef
+    ) {}
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._setUpCarouselConfiguration();
+        this._setUpQuerySubscription();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._setUpInitialValue();
+        this._initialised = true;
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** @hidden */
+    @HostListener('click')
+    onItemClick() {
+        this.activeStateChange.emit();
+    }
+
+    /** @hidden */
+    @HostListener('keydown', ['$event'])
+    keyDownHandler(event: KeyboardEvent): void {
+        if (KeyUtil.isKey(event, 'ArrowDown')) {
+            this.scrollDown();
+            event.preventDefault();
+        } else if (KeyUtil.isKey(event, 'ArrowUp')) {
+            this.scrollUp();
+            event.preventDefault();
+        } else if (KeyUtil.isKey(event, 'ArrowLeft')) {
+            this.focusPreviousColumn.emit();
+        } else if (KeyUtil.isKey(event, 'ArrowRight')) {
+            this.focusNextColumn.emit();
```

Comment:
I think we should also emit this on enter key

Reply:
I'm not sure if it should work like that